mewarpx/mewarpx/diags_store/diag_base.py

```diff
@@ -194,21 +195,26 @@ def __init__(self, diag_steps, preset_string='default',
         if install:
             callbacks.installafterstep(self.text_diag)
 
+        self.particle_steps_total = 0
+        self.previous_particle_steps_total = 0
+
     def text_diag(self):
         """Write requested information to output."""
+        live_parts, parts_per_species_str = self._get_part_nums()
+        self.particle_steps_total += live_parts
+
```

Comment:
This worries me some since we are adding work to every step in the simulation (including a parallel sum). Plus we don't really need this level of resolution on this measurement. I think it'd be fine to just use the `live_parts` present at the `check_timestep() = True` steps and assume the particle count was roughly constant throughout that diagnostic period (so inside the if statement do `self.particle_steps_total += live_parts*steps`).

Reply:
@PhilMiller just fyi ^

Reply:
Also, correct me if I'm wrong but this function does exactly the same as the `update_particle_steps` functions from PR #69, right? So maybe a better way is to pass the text diag to this class and only save the total particle steps in one place.

Reply:
There isn't always a `TextDiag` created during runs though. Perhaps this functionality should just be moved to a common module, maybe `mwxrun`.